src/filter.ts

```diff
@@ -387,41 +387,47 @@ type NotUndefined = string | number | boolean | SomeObject;
  * Given a FilterQuery, returns an object that, when intersected with an Update,
  * marks those properties as required that are guaranteed to exist.
  */
-type RunQuery<Q extends string> = L1Combinations<Q, L1Parts<Q>>;
+type RunQuery<Q extends string> = L1Discriminator<Q, L1Parts<Q>>;
```

Comment:
Combinators were renamed to discriminators, because they are used to narrow down properties. The fact that they can be combined is less speaking.